### DISCUSSION TITLE
Concepts for modularizing vweb applications?

### DISCUSSION THREAD

**Discussion by archandha** [vlang/v#14780] [General]
Hi there,

I recently worked on an large project centered around a Django Rest framework/Python based REST server. In this project we have ~90 REST endpoints, most of them CRUD for different data structures and some of them for result retrieval, configuration etc. As the project is configurable by deployment, one could add or remove some features (endpoints) by not embedding some libraries into the deployment code. 
Obviously, we're not going to do a rewrite in V now but I was playing around with the idea in my spare time if it would be possible one time to implement something like that in V.
As I understand the status quo right now, you have to have the one App struct and derive all endpoints from that. This would need all endpoints to be in the same code module without an easy way skipping this - and no, I don't consider conditional compiling over many code files as easy way. 
Having a struct like the App which gives the context for vweb is great, but attaching every end point function to this one struct limits the options for code organization, doesn't it? How would you approach this? In an class based language you could think of inheriting other classes with endpoints. 

Don't get me wrong, I'm new to the V project, read most of the docs and played around with some code just for fun. I really like the charm of the clean and tight code and would love to see V progress into a state like Go already has

Cheers

**Comment by JalonSolov** [comment]
You can always try https://github.com/nedpals/vex as an alternative to vweb.

**Reply by archandha** [reply]
Hi @JalonSolov thank you for pointing out vex. I'll take a deeper look into this asap.

**Comment by ghost** [comment]
My opinion is that it is very easy to build a web-server and framework from scratch using Soket without using existing vlib net modules.
So it will be easier to create the web server and framework you want if you don't stick to vweb.

**Comment by medvednikov** [comment]
@archandha I agree.

We'll make it possible to use different handlers.

```
`/` => App
`/admin` => Admin
`/user` => User
```

etc

**Reply by archandha** [reply]
Great, that's good news. I already saw vex as an alternative web framework but my point is, most people new to a language will take a look into the std lib first. If this is limited by concept, it might lead people to false conclusions. For the early stage of V I think that is totally ok.  
I think extending these concepts at least on the road map is good to show people how serious the project is about the usage.
